couchdb/server/helpers.py

```diff
@@ -0,0 +1,15 @@
+# -*- coding: utf-8 -*-
+#
+from inspect import getsource
+from textwrap import dedent
+from types import FunctionType
+
+from couchdb import util
+
+
+def maybe_extract_source(fun):
+    if isinstance(fun, FunctionType):
+        return dedent(getsource(fun))
```

Comment:
Oh, right. String source isn't applicable here. I should get more tea at the morning (:
This case is when you pass function object. If it's defined not at the top level (with zero indention), getsource will return it sources with all that indention causing further failures downstream.